src/VisualStudio/Core/Def/Diagnostics/VisualStudioDiagnosticAnalyzerService.cs

```diff
@@ -298,154 +161,156 @@ private void OnRunCodeAnalysisForSelectedProjectStatus(object sender, EventArgs
     private void OnRunCodeAnalysisForSelectedProject(object sender, EventArgs args)
     {
         if (VisualStudioCommandHandlerHelpers.TryGetSelectedProjectHierarchy(_serviceProvider, out var hierarchy))
-        {
             RunAnalyzers(hierarchy);
-        }
     }
 
     public void RunAnalyzers(IVsHierarchy? hierarchy)
     {
-        var project = GetProject(hierarchy);
+        // If a new command comes in to run analyzers again, cancel any existing operation in progress and start a new one.
+        var cancellationToken = _cancellationSeries.CreateNext();
```

Comment:
made it so that running code analysis is actually cancellable.  and that kicking off a new RCA will cancel the last one.